LayoutTests/fast/attachment/resources/attachmentTestUtils.js

```diff
@@ -0,0 +1,20 @@
+function takeScreenshotWhenAttachmentsSettled()
+{
+    console.assert(document.documentElement.classList.contains("reftest-wait"));
+
+    window.addEventListener("load", async function() {
+        let attachments = Array.from(document.getElementsByTagName("attachment"));
+        for (let i = 0; i < 10 + attachments.length; ++i) {
```

Comment:
Why do we stop after an arbitrary limit? Shouldn't we keep polling until the condition (`allIconsLoaded`) is met?

Reply:
Ok, made it an infinite loop. I guess it could catch unexpected never-getting-an-icon cases.